core/include/reboot.h

```diff
@@ -28,7 +30,10 @@
  *
  * This function is used by core_panic() when the DEVELHELP macro is not defined.
  */
-void reboot(void);
+static inline void reboot(void)
+{
+    pm_reboot();
+}
```

Comment:
Hm: I would like to say it with your words: don't leave stuff there because we might need it again at some point in the future, reintroduce once really needed :-)